File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/error/CafeExceptionHandler.java
package io.beaniejoy.dongnecafe.domain.cafe.error;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class CafeExceptionHandler {

    @ExceptionHandler(CafeNotFoundException.class)
    public ResponseEntity<String> handleNotFound(CafeNotFoundException exception) {
        return ResponseEntity.badRequest().body(exception.getMessage());
    }
}

File: dongne-service-api/src/main/java/io/beaniejoy/dongnecafe/domain/cafe/error/CafeExceptionHandler.kt
package io.beaniejoy.dongnecafe.domain.cafe.error

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class CafeExceptionHandler {
    // TODO: error 규격화
    @ExceptionHandler(CafeNotFoundException::class)
    fun handleNotFound(exception: CafeNotFoundException): ResponseEntity<String> {
        return ResponseEntity.badRequest().body(exception.message)
    }
}